sdk/framework/backend.go

```diff
@@ -466,6 +476,16 @@ func (b *Backend) Secret(k string) *Secret {
 	return nil
 }
 
+// WriteSafeReplicationState returns true if this backend instance is capable of writing
+// to storage without receiving an ErrReadOnly error. The active instance in a primary
+// cluster or a local mount on a performance secondary is capable of writing to storage.
+func (b *Backend) WriteSafeReplicationState() bool {
```

Comment:
We could also invert this to something like `HasReadOnlyStorage()` or something similar. This PKI conditional [backend.go#L454](https://github.com/hashicorp/vault/blob/main/builtin/logical/pki/backend.go#L454) is an example of what that could look like.

I'm open either way. Lmk your thoughts!

Reply:
I prefer WriteSafeReplicationState.